File: Sum of All Odd Length Subarrays/Sum of All Odd Length Subarrays.java
//tricky version
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int res=0,n=arr.length;
        for(int i=0;i<n;++i){
            res+=((i+1)*(n-i)+1)/2*arr[i];
        }
        return res;
    }
}

//prefix sum
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int res=0,n=arr.length;
        int[] prefix=new int[n+1];
        for(int i=0;i<n;++i)
            prefix[i+1]=prefix[i]+arr[i];
        for(int i=0;i<n;++i)
            for(int j=i+1;j<=n;j+=2)
                res+=prefix[j]-prefix[i];
        return res;
    }
}

File: Solutions/Sum of All Odd Length Subarrays/Sum of All Odd Length Subarrays.py
#tricky version
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        res=0
        n=len(arr)
        for i,a in enumerate(arr):
            res+=((i+1)*(n-i)+1)//2*a
        return res



#prefix sum
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        n,res=len(arr),0
        prefix=[0]*(n+1)
        for i,a in enumerate(arr):
            prefix[i+1]=prefix[i]+a
        for i in range(0,n):
            for j in range(i+1,n+1,2):
                res+=prefix[j]-prefix[i]
        return res;